platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -167,14 +169,14 @@ def __mount_onie_fs(self):
             self.__umount_onie_fs()
 
         cmd = "fdisk -l | grep 'ONIE boot' | awk '{print $1}'"
-        fs_path = subprocess.check_output(cmd,
-                                          stderr=subprocess.STDOUT,
-                                          shell=True,
-                                          universal_newlines=True).rstrip('\n')
+        cmd1 = ['fdisk', '-l']
+        cmd2 = ['grep', 'ONIE boot']
+        cmd3 = ['awk', '{print $1}']
+        fs_path = check_output_pipe(cmd1, cmd2, cmd3)
 
         os.mkdir(fs_mountpoint)
-        cmd = "mount -n -r -t ext4 {} {}".format(fs_path, fs_mountpoint)
-        subprocess.check_call(cmd, shell=True, universal_newlines=True)
+        cmd = ["mount", "-n", "-r", "-t", "ext4", fs_path, fs_mountpoint]
+        subprocess.check_call(cmd, universal_newlines=True)
```

Comment:
Looks like it is not able to mount fs without `shell=True`.

Reply:
could you share the error log?

Reply:
My bad. it doesn't relevant to the PR. it's because there is a `\n` at the end of `/dev/sda2`.
We will fix it on the platform API side.
Sorry for the inconvenient.

Reply:
Sorry. I realized the missing `\n` was still introduced by this PR. PR #12465 to fix it. can you please review it?
thanks.

Reply:
Thanks for the fix!